aptos-move/framework/src/natives/function_info.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+use aptos_gas_schedule::gas_params::natives::aptos_framework::*;
+use aptos_native_interface::{
+    safely_pop_arg, RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeError,
+    SafeNativeResult,
+};
+use move_binary_format::errors::PartialVMError;
+use move_core_types::{
+    account_address::AccountAddress, gas_algebra::NumBytes, identifier::Identifier,
+    language_storage::ModuleId, vm_status::StatusCode,
+};
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    values::{Reference, StructRef, Value, VectorRef},
+};
+use smallvec::{smallvec, SmallVec};
+use std::collections::VecDeque;
+
+// Extract Identifer from a move value of type &String
+fn identifier_from_ref(v: Value) -> SafeNativeResult<Identifier> {
+    let bytes = v
+        .value_as::<StructRef>()
```

Comment:
Is the move type &String or a struct with a member 0 where the identifier is stored.

Reply:
Yes.